src/order/complete_lattice.lean

```diff
@@ -264,15 +264,27 @@ lemma is_glb.infi_eq (h : is_glb (range s) a) : (⨅j, s j) = a := h.Inf_eq
 theorem le_supr_of_le (i : ι) (h : a ≤ s i) : a ≤ supr s :=
 le_trans h (le_supr _ i)
 
+theorem le_bsupr {p : ι → Prop} {f : Π i, p i → α} (i : ι) (hi : p i) :
+  f i hi ≤ ⨆ i hi, f i hi :=
+le_supr_of_le i $ le_supr (f i) hi
+
 theorem supr_le (h : ∀i, s i ≤ a) : supr s ≤ a :=
 Sup_le $ assume b ⟨i, eq⟩, eq ▸ h i
 
+theorem bsupr_le {p : ι → Prop} {f : Π i (h : p i), α} (h : ∀ i, p i → f i ‹_› ≤ a) :
```

Comment:
`p i → f i ‹_›` is pretty confusing: it's not an implication, it is a `forall`. 
If you change this, then also change all other occurrences of `‹_›`
```suggestion
theorem bsupr_le {p : ι → Prop} {f : Π i (h : p i), α} (h : ∀ i (h : p i) → f i h ≤ a) :
```